### PR TITLE
Lint-free

### LINKED ISSUE #54: Become lint free

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Fixes #54

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -25,9 +25,12 @@
...
         "no-backbone-get-set-outside-model": false,
         "no-implicit-dependencies": [
-            false, // TODO
+            true,
             [
-                "vscode"
+                "vscode",
+                "gulp",
+                "vscode-azureextensiondev",
+                "mocha"
```
Comment (EricJizbaMSFT) [inline tslint.json]: Suggest adding a `tslint.json` file to your test folder that allows all "dev" dependencies

Something like this: https://github.com/microsoft/vscode-azurefunctions/blob/master/test/tslint.json#L5

Comment (StephenWeatherford) [reply tslint.json]: Cool, thanks.

File: src/AzureRMAssets.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export class AzureRMAssets {
 
     // For test dependency injection only
     public static setFunctionsMetadata(metadataString: string): void {
+        // tslint:disable-next-line:typedef
```
Comment (nturinski) [inline src/AzureRMAssets.ts]: Why disable this rule rather than type defining the property??

Comment (StephenWeatherford) [reply src/AzureRMAssets.ts]: Because I hate providing obvious stuff that the compiler is already smart enough to infer anyway.

**Pull request merged**